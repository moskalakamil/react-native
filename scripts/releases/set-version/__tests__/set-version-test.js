/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 * @format
 */

const setVersion = require('../index');
const path = require('path');

describe('setVersion', () => {
  beforeAll(() => {
    jest.mock('path', () => {
      // $FlowIgnore[underconstrained-implicit-instantiation]
      const originalPath = jest.requireActual('path');
      return {
        ...originalPath,
        dirname: () => originalPath.join(__dirname, '__fixtures__/two/levels'),
      };
    });

    jest.mock('fs', () => {
      // $FlowIgnore[underconstrained-implicit-instantiation]
      const originalFs = jest.requireActual('fs');

      return {
        ...originalFs,
        writeFileSync: (packagePath, content) => {
          expect(content).toMatchSnapshot(
            path.basename(path.join(packagePath, '..')),
          );
        },
      };
    });
  });
  test('updates all public packages to version', () => {
    setVersion('0.80.0');
  });
  afterAll(() => {
    jest.unmock('path');
    jest.unmock('fs');
  });
});
