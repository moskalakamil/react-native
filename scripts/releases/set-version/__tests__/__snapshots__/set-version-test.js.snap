// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setVersion updates monorepo for nightly: packages/monorepo-pkg-a/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-a\\",
  \\"version\\": \\"0.81.0-nightly-29282302-abcd1234\\",
  \\"description\\": \\"@monorepo/pkg-a\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-b\\": \\"0.81.0-nightly-29282302-abcd1234\\",
    \\"@monorepo/other\\": \\"0.0.1\\"
  },
  \\"devDependencies\\": {
    \\"@monorepo/pkg-c\\": \\"0.81.0-nightly-29282302-abcd1234\\"
  }
}
"
`;

exports[`setVersion updates monorepo for nightly: packages/monorepo-pkg-b/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-b\\",
  \\"version\\": \\"0.81.0-nightly-29282302-abcd1234\\",
  \\"description\\": \\"@monorepo/pkg-b\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-c\\": \\"0.81.0-nightly-29282302-abcd1234\\",
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for nightly: packages/monorepo-pkg-c/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-c\\",
  \\"version\\": \\"0.81.0-nightly-29282302-abcd1234\\",
  \\"description\\": \\"@monorepo/pkg-c\\",
  \\"dependencies\\": {
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for nightly: packages/react-native/Libraries/Core/ReactNativeVersion.js 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict
 * @generated by scripts/releases/set-rn-version.js
 */

const version: $ReadOnly<{
  major: number,
  minor: number,
  patch: number,
  prerelease: string | null,
}> = {
  major: 0,
  minor: 81,
  patch: 0,
  prerelease: 'nightly-29282302-abcd1234',
};

module.exports = {version};
"
`;

exports[`setVersion updates monorepo for nightly: packages/react-native/React/Base/RCTVersion.m 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

#import \\"RCTVersion.h\\"

NSString* const RCTVersionMajor = @\\"major\\";
NSString* const RCTVersionMinor = @\\"minor\\";
NSString* const RCTVersionPatch = @\\"patch\\";
NSString* const RCTVersionPrerelease = @\\"prerelease\\";


NSDictionary* RCTGetReactNativeVersion(void)
{
  static NSDictionary* __rnVersion;
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^(void){
    __rnVersion = @{
                  RCTVersionMajor: @(0),
                  RCTVersionMinor: @(81),
                  RCTVersionPatch: @(0),
                  RCTVersionPrerelease: @\\"nightly-29282302-abcd1234\\",
                  };
  });
  return __rnVersion;
}
"
`;

exports[`setVersion updates monorepo for nightly: packages/react-native/ReactAndroid/gradle.properties 1`] = `
"VERSION_NAME=0.81.0-nightly-29282302-abcd1234
"
`;

exports[`setVersion updates monorepo for nightly: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/ReactNativeVersion.java 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

package com.facebook.react.modules.systeminfo;

import com.facebook.react.common.MapBuilder;

import java.util.Map;

public class ReactNativeVersion {
  public static final Map<String, Object> VERSION = MapBuilder.<String, Object>of(
      \\"major\\", 0,
      \\"minor\\", 81,
      \\"patch\\", 0,
      \\"prerelease\\", \\"nightly-29282302-abcd1234\\");
}
"
`;

exports[`setVersion updates monorepo for nightly: packages/react-native/ReactCommon/cxxreact/ReactNativeVersion.h 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

#pragma once

#include <cstdint>
#include <string_view>

namespace facebook::react {

constexpr struct {
  int32_t Major = 0;
  int32_t Minor = 81;
  int32_t Patch = 0;
  std::string_view Prerelease = \\"nightly-29282302-abcd1234\\";
} ReactNativeVersion;

} // namespace facebook::react
"
`;

exports[`setVersion updates monorepo for nightly: packages/react-native/package.json 1`] = `
"{
  \\"name\\": \\"react-native\\",
  \\"version\\": \\"0.81.0-nightly-29282302-abcd1234\\",
  \\"description\\": \\"fake react native package\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-a\\": \\"0.81.0-nightly-29282302-abcd1234\\",
    \\"@monorepo/pkg-b\\": \\"0.81.0-nightly-29282302-abcd1234\\",
    \\"@monorepo/pkg-c\\": \\"0.81.0-nightly-29282302-abcd1234\\",
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for nightly: packages/react-native/template/package.json 1`] = `
"{
  \\"name\\": \\"react-native-test-template\\",
  \\"version\\": \\"0.0.1\\",
  \\"private\\": true,
  \\"dependencies\\": {
    \\"react\\": \\"18.2.0\\",
    \\"react-native\\": \\"0.81.0-nightly-29282302-abcd1234\\"
  },
  \\"devDependencies\\": {
    \\"@monorepo/pkg-a\\": \\"0.81.0-nightly-29282302-abcd1234\\",
    \\"@monorepo/pkg-c\\": \\"0.81.0-nightly-29282302-abcd1234\\",
    \\"@types/react\\": \\"^18.2.6\\",
    \\"@types/react-test-renderer\\": \\"^18.0.0\\"
  }
}
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/monorepo-pkg-a/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-a\\",
  \\"version\\": \\"0.80.0-rc.3\\",
  \\"description\\": \\"@monorepo/pkg-a\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-b\\": \\"0.80.0-rc.3\\",
    \\"@monorepo/other\\": \\"0.0.1\\"
  },
  \\"devDependencies\\": {
    \\"@monorepo/pkg-c\\": \\"0.80.0-rc.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/monorepo-pkg-b/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-b\\",
  \\"version\\": \\"0.80.0-rc.3\\",
  \\"description\\": \\"@monorepo/pkg-b\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-c\\": \\"0.80.0-rc.3\\",
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/monorepo-pkg-c/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-c\\",
  \\"version\\": \\"0.80.0-rc.3\\",
  \\"description\\": \\"@monorepo/pkg-c\\",
  \\"dependencies\\": {
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/react-native/Libraries/Core/ReactNativeVersion.js 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict
 * @generated by scripts/releases/set-rn-version.js
 */

const version: $ReadOnly<{
  major: number,
  minor: number,
  patch: number,
  prerelease: string | null,
}> = {
  major: 0,
  minor: 80,
  patch: 0,
  prerelease: 'rc.3',
};

module.exports = {version};
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/react-native/React/Base/RCTVersion.m 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

#import \\"RCTVersion.h\\"

NSString* const RCTVersionMajor = @\\"major\\";
NSString* const RCTVersionMinor = @\\"minor\\";
NSString* const RCTVersionPatch = @\\"patch\\";
NSString* const RCTVersionPrerelease = @\\"prerelease\\";


NSDictionary* RCTGetReactNativeVersion(void)
{
  static NSDictionary* __rnVersion;
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^(void){
    __rnVersion = @{
                  RCTVersionMajor: @(0),
                  RCTVersionMinor: @(80),
                  RCTVersionPatch: @(0),
                  RCTVersionPrerelease: @\\"rc.3\\",
                  };
  });
  return __rnVersion;
}
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/react-native/ReactAndroid/gradle.properties 1`] = `
"VERSION_NAME=0.80.0-rc.3
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/ReactNativeVersion.java 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

package com.facebook.react.modules.systeminfo;

import com.facebook.react.common.MapBuilder;

import java.util.Map;

public class ReactNativeVersion {
  public static final Map<String, Object> VERSION = MapBuilder.<String, Object>of(
      \\"major\\", 0,
      \\"minor\\", 80,
      \\"patch\\", 0,
      \\"prerelease\\", \\"rc.3\\");
}
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/react-native/ReactCommon/cxxreact/ReactNativeVersion.h 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

#pragma once

#include <cstdint>
#include <string_view>

namespace facebook::react {

constexpr struct {
  int32_t Major = 0;
  int32_t Minor = 80;
  int32_t Patch = 0;
  std::string_view Prerelease = \\"rc.3\\";
} ReactNativeVersion;

} // namespace facebook::react
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/react-native/package.json 1`] = `
"{
  \\"name\\": \\"react-native\\",
  \\"version\\": \\"0.80.0-rc.3\\",
  \\"description\\": \\"fake react native package\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-a\\": \\"0.80.0-rc.3\\",
    \\"@monorepo/pkg-b\\": \\"0.80.0-rc.3\\",
    \\"@monorepo/pkg-c\\": \\"0.80.0-rc.3\\",
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for release-candidate: packages/react-native/template/package.json 1`] = `
"{
  \\"name\\": \\"react-native-test-template\\",
  \\"version\\": \\"0.0.1\\",
  \\"private\\": true,
  \\"dependencies\\": {
    \\"react\\": \\"18.2.0\\",
    \\"react-native\\": \\"0.80.0-rc.3\\"
  },
  \\"devDependencies\\": {
    \\"@monorepo/pkg-a\\": \\"0.80.0-rc.3\\",
    \\"@monorepo/pkg-c\\": \\"0.80.0-rc.3\\",
    \\"@types/react\\": \\"^18.2.6\\",
    \\"@types/react-test-renderer\\": \\"^18.0.0\\"
  }
}
"
`;

exports[`setVersion updates monorepo for stable version: packages/monorepo-pkg-a/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-a\\",
  \\"version\\": \\"0.80.1\\",
  \\"description\\": \\"@monorepo/pkg-a\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-b\\": \\"0.80.1\\",
    \\"@monorepo/other\\": \\"0.0.1\\"
  },
  \\"devDependencies\\": {
    \\"@monorepo/pkg-c\\": \\"0.80.1\\"
  }
}
"
`;

exports[`setVersion updates monorepo for stable version: packages/monorepo-pkg-b/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-b\\",
  \\"version\\": \\"0.80.1\\",
  \\"description\\": \\"@monorepo/pkg-b\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-c\\": \\"0.80.1\\",
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for stable version: packages/monorepo-pkg-c/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-c\\",
  \\"version\\": \\"0.80.1\\",
  \\"description\\": \\"@monorepo/pkg-c\\",
  \\"dependencies\\": {
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for stable version: packages/react-native/Libraries/Core/ReactNativeVersion.js 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict
 * @generated by scripts/releases/set-rn-version.js
 */

const version: $ReadOnly<{
  major: number,
  minor: number,
  patch: number,
  prerelease: string | null,
}> = {
  major: 0,
  minor: 80,
  patch: 1,
  prerelease: null,
};

module.exports = {version};
"
`;

exports[`setVersion updates monorepo for stable version: packages/react-native/React/Base/RCTVersion.m 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

#import \\"RCTVersion.h\\"

NSString* const RCTVersionMajor = @\\"major\\";
NSString* const RCTVersionMinor = @\\"minor\\";
NSString* const RCTVersionPatch = @\\"patch\\";
NSString* const RCTVersionPrerelease = @\\"prerelease\\";


NSDictionary* RCTGetReactNativeVersion(void)
{
  static NSDictionary* __rnVersion;
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^(void){
    __rnVersion = @{
                  RCTVersionMajor: @(0),
                  RCTVersionMinor: @(80),
                  RCTVersionPatch: @(1),
                  RCTVersionPrerelease: [NSNull null],
                  };
  });
  return __rnVersion;
}
"
`;

exports[`setVersion updates monorepo for stable version: packages/react-native/ReactAndroid/gradle.properties 1`] = `
"VERSION_NAME=0.80.1
"
`;

exports[`setVersion updates monorepo for stable version: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/ReactNativeVersion.java 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

package com.facebook.react.modules.systeminfo;

import com.facebook.react.common.MapBuilder;

import java.util.Map;

public class ReactNativeVersion {
  public static final Map<String, Object> VERSION = MapBuilder.<String, Object>of(
      \\"major\\", 0,
      \\"minor\\", 80,
      \\"patch\\", 1,
      \\"prerelease\\", null);
}
"
`;

exports[`setVersion updates monorepo for stable version: packages/react-native/ReactCommon/cxxreact/ReactNativeVersion.h 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

#pragma once

#include <cstdint>
#include <string_view>

namespace facebook::react {

constexpr struct {
  int32_t Major = 0;
  int32_t Minor = 80;
  int32_t Patch = 1;
  std::string_view Prerelease = \\"\\";
} ReactNativeVersion;

} // namespace facebook::react
"
`;

exports[`setVersion updates monorepo for stable version: packages/react-native/package.json 1`] = `
"{
  \\"name\\": \\"react-native\\",
  \\"version\\": \\"0.80.1\\",
  \\"description\\": \\"fake react native package\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-a\\": \\"0.80.1\\",
    \\"@monorepo/pkg-b\\": \\"0.80.1\\",
    \\"@monorepo/pkg-c\\": \\"0.80.1\\",
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo for stable version: packages/react-native/template/package.json 1`] = `
"{
  \\"name\\": \\"react-native-test-template\\",
  \\"version\\": \\"0.0.1\\",
  \\"private\\": true,
  \\"dependencies\\": {
    \\"react\\": \\"18.2.0\\",
    \\"react-native\\": \\"0.80.1\\"
  },
  \\"devDependencies\\": {
    \\"@monorepo/pkg-a\\": \\"0.80.1\\",
    \\"@monorepo/pkg-c\\": \\"0.80.1\\",
    \\"@types/react\\": \\"^18.2.6\\",
    \\"@types/react-test-renderer\\": \\"^18.0.0\\"
  }
}
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/monorepo-pkg-a/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-a\\",
  \\"version\\": \\"0.82.0-main\\",
  \\"description\\": \\"@monorepo/pkg-a\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-b\\": \\"0.82.0-main\\",
    \\"@monorepo/other\\": \\"0.0.1\\"
  },
  \\"devDependencies\\": {
    \\"@monorepo/pkg-c\\": \\"0.82.0-main\\"
  }
}
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/monorepo-pkg-b/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-b\\",
  \\"version\\": \\"0.82.0-main\\",
  \\"description\\": \\"@monorepo/pkg-b\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-c\\": \\"0.82.0-main\\",
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/monorepo-pkg-c/package.json 1`] = `
"{
  \\"name\\": \\"@monorepo/pkg-c\\",
  \\"version\\": \\"0.82.0-main\\",
  \\"description\\": \\"@monorepo/pkg-c\\",
  \\"dependencies\\": {
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/react-native/Libraries/Core/ReactNativeVersion.js 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict
 * @generated by scripts/releases/set-rn-version.js
 */

const version: $ReadOnly<{
  major: number,
  minor: number,
  patch: number,
  prerelease: string | null,
}> = {
  major: 1000,
  minor: 0,
  patch: 0,
  prerelease: null,
};

module.exports = {version};
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/react-native/React/Base/RCTVersion.m 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

#import \\"RCTVersion.h\\"

NSString* const RCTVersionMajor = @\\"major\\";
NSString* const RCTVersionMinor = @\\"minor\\";
NSString* const RCTVersionPatch = @\\"patch\\";
NSString* const RCTVersionPrerelease = @\\"prerelease\\";


NSDictionary* RCTGetReactNativeVersion(void)
{
  static NSDictionary* __rnVersion;
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^(void){
    __rnVersion = @{
                  RCTVersionMajor: @(1000),
                  RCTVersionMinor: @(0),
                  RCTVersionPatch: @(0),
                  RCTVersionPrerelease: [NSNull null],
                  };
  });
  return __rnVersion;
}
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/react-native/ReactAndroid/gradle.properties 1`] = `
"VERSION_NAME=1000.0.0
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/ReactNativeVersion.java 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

package com.facebook.react.modules.systeminfo;

import com.facebook.react.common.MapBuilder;

import java.util.Map;

public class ReactNativeVersion {
  public static final Map<String, Object> VERSION = MapBuilder.<String, Object>of(
      \\"major\\", 1000,
      \\"minor\\", 0,
      \\"patch\\", 0,
      \\"prerelease\\", null);
}
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/react-native/ReactCommon/cxxreact/ReactNativeVersion.h 1`] = `
"/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @generated by scripts/releases/set-rn-version.js
 */

#pragma once

#include <cstdint>
#include <string_view>

namespace facebook::react {

constexpr struct {
  int32_t Major = 1000;
  int32_t Minor = 0;
  int32_t Patch = 0;
  std::string_view Prerelease = \\"\\";
} ReactNativeVersion;

} // namespace facebook::react
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/react-native/package.json 1`] = `
"{
  \\"name\\": \\"react-native\\",
  \\"version\\": \\"1000.0.0\\",
  \\"description\\": \\"fake react native package\\",
  \\"dependencies\\": {
    \\"@monorepo/pkg-a\\": \\"0.82.0-main\\",
    \\"@monorepo/pkg-b\\": \\"0.82.0-main\\",
    \\"@monorepo/pkg-c\\": \\"0.82.0-main\\",
    \\"metro-config\\": \\"^0.80.3\\",
    \\"metro-runtime\\": \\"^0.80.3\\"
  }
}
"
`;

exports[`setVersion updates monorepo on main after release cut: packages/react-native/template/package.json 1`] = `
"{
  \\"name\\": \\"react-native-test-template\\",
  \\"version\\": \\"0.0.1\\",
  \\"private\\": true,
  \\"dependencies\\": {
    \\"react\\": \\"18.2.0\\",
    \\"react-native\\": \\"1000.0.0\\"
  },
  \\"devDependencies\\": {
    \\"@monorepo/pkg-a\\": \\"0.82.0-main\\",
    \\"@monorepo/pkg-c\\": \\"0.82.0-main\\",
    \\"@types/react\\": \\"^18.2.6\\",
    \\"@types/react-test-renderer\\": \\"^18.0.0\\"
  }
}
"
`;
